examples/z_put.c

```diff
@@ -23,6 +23,9 @@ int main(int argc, char **argv) {
     if (argc > 1) keyexpr = argv[1];
     if (argc > 2) value = argv[2];
 
+    z_owned_bytes_map_t attachements = z_bytes_map_new();
+    z_bytes_map_insert_by_copy(&attachements, z_bytes_new("hello"), z_bytes_new("there"));
```

Comment:
1. Is providing raw bytes supported or should it be only strings?
2. is it possible to provide the buffer and not allocate buffers ? i would like to reduce the number of allocations during run time

Reply:
1. Yes, `z_bytes_new` is just a constructor for a bytes view from a null terminated string, but you may construct `z_bytes_t` in other ways, aliasing whatever memory you like.
2. It is possible:
  - The map itself is able to alias buffers rather than copy them, but it's still a map that will allocate as part of its functioning.
  - `z_attachment_t` is what actually provides Zenoh with the data to send, and does so by being a v-tabled iterator over aliased byte views. If you like, you can provide that v-table for any type you want, allowing you to have 0 need for allocations in the whole process.

Reply:
thanks @p-avital  can you please explain why do you need a map?
What not just provide a way to provide a pointer to a header buffer ?

Reply:
The idea is to provide a flexible way for the user to provide metadata to be attached to the payload. Having one single pointer to a buffer forces the user to potentially allocate a buffer and serialize himself multiple metadata in it first. From a usability perspective, we believe a key-value semantics would be more flexible so as to cover more diverse use cases.

Reply:
This is because the abstraction for metadata is a map of byte view to byte view. Notably because we've seen the need for multiple metadata fields, and that a single buffer would force the user to serialise their metadata on their own before we serialise it again into a packet. This would be both wasteful and less user friendly. This is why we chose this specific abstraction.

You can choose to have a single buffer and have they key to it be an empty slice, the iterator for that wouldn't be hard to write, and would never need to allocate.